scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -677,26 +684,34 @@ private void createCertificateRequest(final ClientHello clientHello, final DTLSF
 	private byte[] receivedClientKeyExchange(final PSKClientKeyExchange message) throws HandshakeException {
 
 		clientKeyExchange = message;
-		byte[] psk = null;
 
-		// use the client's PSK identity to get right preshared key
+		// use the client's PSK identity to look up the pre-shared key
 		String identity = message.getIdentity();
-
-		LOGGER.debug("Client [{}] uses PSK identity [{}]",
-				new Object[]{getPeerAddress(), identity});
+		byte[] psk = pskStore.getKey(getIndicatedServerNames(), identity);
+		String virtualHost = null;
 
 		if (getIndicatedServerNames() == null) {
-			psk = pskStore.getKey(identity);
+			LOGGER.debug("client [{}] uses PSK identity [{}]", getPeerAddress(), identity);
 		} else {
-			psk = pskStore.getKey(getIndicatedServerNames(), identity);
+			// SNI is enabled and the client has indicated a virtual host
+			ServerName serverName = getIndicatedServerNames().getServerName(NameType.HOST_NAME);
```

Comment:
Multiple `getIndicatedServerNames()` maybe clearer, if get once and stored in a local variable.